lib/ansible/module_utils/basic.py

```diff
@@ -743,6 +743,7 @@ def __init__(self, argument_spec, bypass_checks=False, no_log=False,
 
         # This is for backwards compatibility only.
         self._CHECK_ARGUMENT_TYPES_DISPATCHER = DEFAULT_TYPE_VALIDATORS
+        self._CHECK_ARGUMENT_TYPES_DISPATCHER['str_strict'] = self._check_type_str_strict
```

Comment:
Since `DEFAULT_TYPE_VALIDATORS` moved out of `AnsibleModule`, I had to put this in to make `str_strict` work. I don't particularly like adding a private method to a constant where all other entries are standalone functions.

Since "strict string" checking is really just special call and handling to `check_type_str()`, maybe this should move to a standalone function as well.